20241014-wordcloud/README.md

```diff
@@ -0,0 +1,91 @@
+# 14 Oct 2024, wordcloud (SvelteKit / TypeScript / PostgresQL)
+
+**Original source: https://github.com/penelopeysm/wordcloud**
+
+## Description
+
+An web app that lets multiple users input text and shows a word cloud of everyone's input, written in SvelteKit and is backed by a PostgresQL database which stores the actual inputs.
+
+The quality of the wordcloud generation is not amazing, the library that I used here isn't super great.
+Suggestions are welcome :)
+
+You can view the deployed web app at https://wordcloud.pysm.dev/.
+This was used to create the word cloud of languages and technologies on the [new REG website](https://alan-turing-institute.github.io/REG/).
+(I've since set up a new table, to avoid 'contaminating' the data there.)
+
+## Running locally
+
+To run this locally, you will need to do the following steps:
+
+1. Install PostgresQL if you don't have it:
+
+   ```bash
+   brew install postgresql
+   ```
+
+1. Create a new folder for the database, and launch a PostgresQL server in that folder:
+
+   ```bash
+   mkdir db
+   cd db
+   initdb .
+   pg_ctl start -D .
+   ```
+
+1. While you're still in the `db` directory, create a new database called `wordcloud`, then drop into the PostgresQL shell:
+
+   ```bash
+   createdb wordcloud
+   psql wordcloud
+   ```
+
+1. You should see a prompt that looks like this.
+
+   ```
+   $ psql wordcloud
+   psql (14.13 (Homebrew))
+   Type "help" for help.
+
+   wordcloud=#
+   ```
+
+1. Inside this prompt, create a new table by pasting the following text, then pressing `Enter`.
+
+   ```sql
+   CREATE TABLE words (item VARCHAR(255) PRIMARY KEY, count INT NOT NULL);
```

Comment:
For ease of deployment, I think you should include a `create_words_table.sql`  file with the SQL scripts. Or, if possible, a `create_database.sh` script for setting up the whole PostgreSQL db.